tests/ut/codegen/test_phase_fence_dep_compression.py

```
@@ -280,6 +291,88 @@ def main(
+            ) -> pl.Tensor[[64], pl.FP32]:
+                return out
+
+            @pl.function(type=pl.FunctionType.AIV)
+            def consume(self, x: pl.Tensor[[64], pl.FP32]) -> pl.Tensor[[64], pl.FP32]:
+                return x
+
+            @pl.function(type=pl.FunctionType.Orchestration)
+            def main(self, scratch: pl.Tensor[[64], pl.FP32]) -> pl.Tensor[[64], pl.FP32]:
+                carried = scratch
+                last_tid = pl.system.task_invalid()
+                for _i, (carried, last_tid) in pl.range(
```

Ruff (0.15.20)

[warning] Loop control variable `carried` overrides iterable it iterates

(B020)